File: libs/rs-sdk-integration-tests/src/vm_tests.rs
use seda_sdk_rs::{bytes::ToBytes, http::http_fetch, process::Process};

pub fn test_tally_vm_mode() {
    if Process::is_dr_vm_mode() {
        Process::error(&"dr".to_bytes());
    } else {
        Process::success(&"tally".to_bytes());
    }
}

pub fn test_tally_vm_http() {
    let response = http_fetch(
        "https://api.binance.com/api/v3/ticker/price?symbol=eth-usdt",
        None,
    );

    if response.is_ok() {
        Process::error(&"this should not be allowed in tally mode".to_bytes());
        return;
    }

    Process::success(&response.bytes);
}

File: libs/go-sdk-integration-tests/tests/vm-tests.go
package tests

import (
	"github.com/seda-protocol/seda-sdk/http"
	"github.com/seda-protocol/seda-sdk/process"
)

func TestTallyVmMode() {
	if process.IsDrVmMode() {
		process.Error([]byte("dr"))
	} else {
		process.Success([]byte("tally"))
	}
}

func TestTallyVmHttp() {
	response := http.HttpFetch("https://api.binance.com/api/v3/ticker/price?symbol=eth-usdt", nil)

	if response.IsOk() {
		process.Error([]byte("this should not be allowed in tally mode"))
		return
	}

	process.Success(response.Bytes)
}
